Sources/PackageModel/UserToolchain.swift

```diff
@@ -499,6 +502,18 @@ public final class UserToolchain: Toolchain {
         self.isSwiftDevelopmentToolchain = false
         #endif
 
+        if let customInstalledSwiftPMConfiguration = customInstalledSwiftPMConfiguration {
+            self.installedSwiftPMConfiguration = customInstalledSwiftPMConfiguration
+        } else {
+            let path = self.swiftCompilerPath.parentDirectory.parentDirectory.appending(components: ["share", "pm", "config.json"])
+            if localFileSystem.exists(path) {
+                self.installedSwiftPMConfiguration = try JSONDecoder.makeWithDefaults().decode(path: path, fileSystem: localFileSystem, as: InstalledSwiftPMConfiguration.self)
+            } else {
+                // We *could* eventually make this an error, but not for a few releases.
+                self.installedSwiftPMConfiguration = InstalledSwiftPMConfiguration(version: 0, swiftSyntaxVersionForMacroTemplate: .init(major: 509, minor: 0, patch: 0))
```

Comment:
Shouldn’t we use `.default` here and set `509.0.0` in the default?

Reply:
That could make sense, yah.